File: sumMixedArray.js
function sumMix(x){
  console.log(x);
  for(var i = 0; i < x.length; i++) {
    //console.log(typeof x[i]);
    let current = x[i];
    if (typeof current == 'string') {
      // console.log(x[i]);
      let num = Number(current);
      // console.log(x[i]);
      x.splice(current, 1);
      // console.log(x[i]);
      //console.log(x[i]);
    //   console.log('string')
    // } else {
    //   console.log('number')
    }
    console.log(x);
  }

  //console.log(x);
  //newArray = x.forEach(element => console.log(element));//parseInt(element, 10));
  //console.log(newArray);
}

sumMix([9, 3, '7', '3']);

//22

//a totally separate day
function sumMix(x){
  return x.map(a => parseInt(a)).reduce((sum, num) => sum + num, 0);
}

File: sum_mixed_array.rb
#Feb 10 2020
def sumMix(x)
  for i in x do
    # if x[i].class
    # if x[i].is_a? String
    #
    # end

    curr = x[i]
    puts "#{curr}"
    #printf x[i]

    # puts x[0]
    # puts x[1]
    # puts x[2]
    # puts x[3]

    # if (x[i]).is_a? String
    #   puts "String"
    # end
    # x.map { |i| x[i].is_a? String ? x[i] : x[i].to_i }
  end
  # return x
end

sumMix([9, 3, '7', '3'])


# p array.map { |x| x == 4 ? 'Z' : x }

# => [1, 2, 3, 'Z']

#Aug 2 2020

def sum_mix(x)
  return x.map { |a| a.to_i }.reduce { |sum, num| sum + num }
end

def sum_mix(x)
  x.sum(&:to_i)
end
